### PR TITLE
added variable equality conditions

### PR DESCRIPTION
**Pull request by bbeck13**
Using the power of regex and silly code you can now do equality conditions on dd variables

### CODE REVIEW

**Comment by bbeck13** [comment]
```
mvn install
```
```
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for com.csc530:sat:jar:0.0.1
[WARNING] 'dependencies.dependency.systemPath' for jsmtlib:com.jsmtlib:jar should not point at files within the project directory, ${project.basedir}/jSMTLIB/jSMTLIB.jar will be unresolvable by dependent projects @ line 47, column 22
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 81, column 19
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building sat 0.0.1
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ sat ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/badmin/Documents/School/S17/530/530-SAT/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ sat ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 21 source files to /home/badmin/Documents/School/S17/530/530-SAT/target/classes
[WARNING] /home/badmin/Documents/School/S17/530/530-SAT/src/main/java/com/csc530/sat/branch/Decision.java: /home/badmin/Documents/School/S17/530/530-SAT/src/main/java/com/csc530/sat/branch/Decision.java uses unchecked or unsafe operations.
[WARNING] /home/badmin/Documents/School/S17/530/530-SAT/src/main/java/com/csc530/sat/branch/Decision.java: Recompile with -Xlint:unchecked for details.
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ sat ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/badmin/Documents/School/S17/530/530-SAT/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ sat ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 6 source files to /home/badmin/Documents/School/S17/530/530-SAT/target/test-classes
[INFO]
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ sat ---
[INFO] Surefire report directory: /home/badmin/Documents/School/S17/530/530-SAT/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.csc530.sat.BooleanTheoryTest
{"Variable(name=isItRaining?, valueClass=class java.lang.Boolean)": {"true": {"Variable(name=iHaveAnUmbrella, valueClass=class java.lang.Boolean)": {"true": "SATISFIABLE", "false": "UNSATISFIABLE"}}, "false": "SATISFIABLE"}}
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.265 sec
Running com.csc530.sat.AppTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec
Running com.csc530.sat.IntegerTheoryTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Running com.csc530.sat.VariableEQTheoryTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.019 sec
Running com.csc530.cnf.CNFParserTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.054 sec
Running com.csc530.smtlib.SmtLibRunnerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.127 sec

Results :

Tests run: 14, Failures: 0, Errors: 0, Skipped: 0

[INFO]
[INFO] --- maven-jar-plugin:2.2:jar (default-jar) @ sat ---
[INFO] Building jar: /home/badmin/Documents/School/S17/530/530-SAT/target/sat-0.0.1.jar
[INFO]
[INFO] --- maven-assembly-plugin:2.2-beta-4:single (default) @ sat ---
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] org/ already added, skipping
[INFO] org/apache/ already added, skipping
[INFO] org/apache/commons/ already added, skipping
[INFO] META-INF/LICENSE.txt already added, skipping
[INFO] META-INF/NOTICE.txt already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] META-INF/maven/org.apache.commons/ already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] org/ already added, skipping
[INFO] org/apache/ already added, skipping
[INFO] org/apache/commons/ already added, skipping
[INFO] META-INF/NOTICE.txt already added, skipping
[INFO] META-INF/LICENSE.txt already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] com/ already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] com/csc530/sat/condition/VariableNEQ.class already added, skipping
[INFO] com/csc530/sat/condition/VariableNEQ$1.class already added, skipping
[INFO] com/csc530/sat/condition/integer/IntegerNEQCondition.class already added, skipping
[INFO] com/csc530/sat/condition/integer/IntegerEQCondition.class already added, skipping
[INFO] com/csc530/sat/condition/bool/BooleanCondition$BooleanConditionBuilder.class already added, skipping
[INFO] com/csc530/sat/condition/bool/BooleanCondition.class already added, skipping
[INFO] com/csc530/sat/condition/VariableEQ.class already added, skipping
[INFO] com/csc530/sat/condition/VariableEQ$2.class already added, skipping
[INFO] com/csc530/sat/condition/DDCondition.class already added, skipping
[INFO] com/csc530/sat/condition/VariableEQ$1.class already added, skipping
[INFO] com/csc530/sat/condition/VariableNEQ$2.class already added, skipping
[INFO] com/csc530/sat/App.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagramLeaf.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagram.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagramNode.class already added, skipping
[INFO] com/csc530/sat/Variable.class already added, skipping
[INFO] com/csc530/sat/branch/Decision.class already added, skipping
[INFO] com/csc530/sat/branch/Decision$DecisionBuilder.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagramNode$1.class already added, skipping
[INFO] com/csc530/sat/type/integer/IntegerDDType.class already added, skipping
[INFO] com/csc530/sat/type/bool/BooleanDDType.class already added, skipping
[INFO] com/csc530/sat/type/bool/BooleanDDType$1.class already added, skipping
[INFO] com/csc530/sat/type/bool/BooleanDDType$2.class already added, skipping
[INFO] com/csc530/sat/type/DDType.class already added, skipping
[INFO] com/csc530/sat/type/IntegerPair.class already added, skipping
[INFO] com/csc530/sat/util/RandomValues.class already added, skipping
[INFO] com/csc530/sat/Variable$VariableBuilder.class already added, skipping
[INFO] com/csc530/cnf/CNFParser.class already added, skipping
[INFO] com/csc530/smtlib/SmtLibRunner.class already added, skipping
[INFO] com/csc530/smtlib/ExpressionVisitor.class already added, skipping
[INFO] com/csc530/smtlib/SimpleSolver.class already added, skipping
[INFO] Building jar: /home/badmin/Documents/School/S17/530/530-SAT/target/sat-0.0.1-jar-with-dependencies.jar
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] org/ already added, skipping
[INFO] org/apache/ already added, skipping
[INFO] org/apache/commons/ already added, skipping
[INFO] META-INF/LICENSE.txt already added, skipping
[INFO] META-INF/NOTICE.txt already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] META-INF/maven/org.apache.commons/ already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] org/ already added, skipping
[INFO] org/apache/ already added, skipping
[INFO] org/apache/commons/ already added, skipping
[INFO] META-INF/NOTICE.txt already added, skipping
[INFO] META-INF/LICENSE.txt already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] META-INF/ already added, skipping
[INFO] META-INF/MANIFEST.MF already added, skipping
[INFO] com/ already added, skipping
[INFO] META-INF/maven/ already added, skipping
[INFO] com/csc530/sat/condition/VariableNEQ.class already added, skipping
[INFO] com/csc530/sat/condition/VariableNEQ$1.class already added, skipping
[INFO] com/csc530/sat/condition/integer/IntegerNEQCondition.class already added, skipping
[INFO] com/csc530/sat/condition/integer/IntegerEQCondition.class already added, skipping
[INFO] com/csc530/sat/condition/bool/BooleanCondition$BooleanConditionBuilder.class already added, skipping
[INFO] com/csc530/sat/condition/bool/BooleanCondition.class already added, skipping
[INFO] com/csc530/sat/condition/VariableEQ.class already added, skipping
[INFO] com/csc530/sat/condition/VariableEQ$2.class already added, skipping
[INFO] com/csc530/sat/condition/DDCondition.class already added, skipping
[INFO] com/csc530/sat/condition/VariableEQ$1.class already added, skipping
[INFO] com/csc530/sat/condition/VariableNEQ$2.class already added, skipping
[INFO] com/csc530/sat/App.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagramLeaf.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagram.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagramNode.class already added, skipping
[INFO] com/csc530/sat/Variable.class already added, skipping
[INFO] com/csc530/sat/branch/Decision.class already added, skipping
[INFO] com/csc530/sat/branch/Decision$DecisionBuilder.class already added, skipping
[INFO] com/csc530/sat/DecisionDiagramNode$1.class already added, skipping
[INFO] com/csc530/sat/type/integer/IntegerDDType.class already added, skipping
[INFO] com/csc530/sat/type/bool/BooleanDDType.class already added, skipping
[INFO] com/csc530/sat/type/bool/BooleanDDType$1.class already added, skipping
[INFO] com/csc530/sat/type/bool/BooleanDDType$2.class already added, skipping
[INFO] com/csc530/sat/type/DDType.class already added, skipping
[INFO] com/csc530/sat/type/IntegerPair.class already added, skipping
[INFO] com/csc530/sat/util/RandomValues.class already added, skipping
[INFO] com/csc530/sat/Variable$VariableBuilder.class already added, skipping
[INFO] com/csc530/cnf/CNFParser.class already added, skipping
[INFO] com/csc530/smtlib/SmtLibRunner.class already added, skipping
[INFO] com/csc530/smtlib/ExpressionVisitor.class already added, skipping
[INFO] com/csc530/smtlib/SimpleSolver.class already added, skipping
[INFO]
[INFO] --- maven-install-plugin:2.4:install (default-install) @ sat ---
[INFO] Installing /home/badmin/Documents/School/S17/530/530-SAT/target/sat-0.0.1.jar to /home/badmin/.m2/repository/com/csc530/sat/0.0.1/sat-0.0.1.jar
[INFO] Installing /home/badmin/Documents/School/S17/530/530-SAT/pom.xml to /home/badmin/.m2/repository/com/csc530/sat/0.0.1/sat-0.0.1.pom
[INFO] Installing /home/badmin/Documents/School/S17/530/530-SAT/target/sat-0.0.1-jar-with-dependencies.jar to /home/badmin/.m2/repository/com/csc530/sat/0.0.1/sat-0.0.1-jar-with-dependencies.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.343 s
[INFO] Finished at: 2017-06-05T23:37:39-07:00
[INFO] Final Memory: 26M/195M
[INFO] ------------------------------------------------------------------------
```

File: src/main/java/com/csc530/sat/DecisionDiagramNode.java [hunk 1]
```diff
@@ -23,19 +31,43 @@
...
+    private static final Pattern combination = Pattern
+            .compile("^([a-zA-Z0-9]+)_-_([a-zA-Z0-9]+)$");
     private final Variable<T> primaryVariable;
     private final LinkedHashSet<Variable> variables;
     private final Decision<T> branches;
 
     public static <T> DecisionDiagramNode<T> of(final Variable<T> variable,
             final DDCondition<T> condition) {
-        LinkedHashSet<Variable<?>> vars = new LinkedHashSet<>();
+        LinkedHashSet<Variable> vars = new LinkedHashSet<>();
```
Comment (avilanicolas) [inline src/main/java/com/csc530/sat/DecisionDiagramNode.java]: 👍

File: src/main/java/com/csc530/sat/DecisionDiagramNode.java [hunk 2]
```diff
@@ -121,13 +153,38 @@ public boolean satisfies(final Map<Variable, DDType> assignment) {
...
+                        .anyMatch(val -> val.getName().equals(var1)) &&
+                        assignment.entrySet().stream().map(Entry::getKey)
+                                .anyMatch(val -> val.getName().equals(var2))) {
+                    Variable one = assignment.entrySet().stream().map(Entry::getKey)
+                            .filter(val -> val.getName().equals(var1)).findFirst().get();
+                    Variable two = assignment.entrySet().stream().map(Entry::getKey)
+                            .filter(val -> val.getName().equals(var2)).findFirst().get();
+                    branch = branches.getBranch(new DDType() {
+                        @Override
+                        public Pair<DDType, DDType> getValue() {
```
Comment (avilanicolas) [inline src/main/java/com/csc530/sat/DecisionDiagramNode.java]: 💩 💩

File: src/main/java/com/csc530/sat/DecisionDiagramNode.java [hunk 3]
```diff
@@ -121,13 +153,38 @@ public boolean satisfies(final Map<Variable, DDType> assignment) {
...
-        // Or if it doesn't contain this node's variable!
+        DecisionDiagram branch;
+        // = branches.getBranch(assignment.get(primaryVariable));
         if (!assignment.containsKey(primaryVariable)) {
-            return false;
+            Matcher matcher = combination.matcher(primaryVariable.getName());
+            if (matcher.matches()) {
+                String var1 = matcher.group(1);
+                String var2 = matcher.group(2);
+                if (assignment.entrySet().stream().map(Entry::getKey)
```
Comment (avilanicolas) [inline src/main/java/com/csc530/sat/DecisionDiagramNode.java]: put this in a fucking variable you goddamn pacy fuck oh I mean pineapple 🅱️

**Pull request merged**